tests/workflows/test_from_csv_to_parquet.py

```diff
@@ -0,0 +1,143 @@
+        ("Actor1Type1Code", "string[pyarrow]"),
+        ("Actor1Type2Code", "string[pyarrow]"),
+        ("Actor1Type3Code", "string[pyarrow]"),
+        ("Actor2Code", "string[pyarrow]"),
+        ("Actor2Name", "string[pyarrow]"),
+        ("Actor2CountryCode", "string[pyarrow]"),
+        ("Actor2KnownGroupCode", "string[pyarrow]"),
+        ("Actor2EthnicCode", "string[pyarrow]"),
+        ("Actor2Religion1Code", "string[pyarrow]"),
+        ("Actor2Religion2Code", "string[pyarrow]"),
+        ("Actor2Type1Code", "string[pyarrow]"),
+        ("Actor2Type2Code", "string[pyarrow]"),
+        ("Actor2Type3Code", "string[pyarrow]"),
+        ("IsRootEvent", "Int64"),
+        ("EventCode", "string[pyarrow]"),
+        ("EventBaseCode", "string[pyarrow]"),
+        ("EventRootCode", "string[pyarrow]"),
+        ("QuadClass", "Int64"),
+        ("GoldsteinScale", "Float64"),
+        ("NumMentions", "Int64"),
+        ("NumSources", "Int64"),
+        ("NumArticles", "Int64"),
+        ("AvgTone", "Float64"),
+        ("Actor1Geo_Type", "Int64"),
+        ("Actor1Geo_Fullname", "string[pyarrow]"),
+        ("Actor1Geo_CountryCode", "string[pyarrow]"),
+        ("Actor1Geo_ADM1Code", "string[pyarrow]"),
+        ("Actor1Geo_Lat", "Float64"),
+        ("Actor1Geo_Long", "Float64"),
+        ("Actor1Geo_FeatureID", "string[pyarrow]"),
+        ("Actor2Geo_Type", "Int64"),
+        ("Actor2Geo_Fullname", "string[pyarrow]"),
+        ("Actor2Geo_CountryCode", "string[pyarrow]"),
+        ("Actor2Geo_ADM1Code", "string[pyarrow]"),
+        ("Actor2Geo_Lat", "Float64"),
+        ("Actor2Geo_Long", "Float64"),
+        ("Actor2Geo_FeatureID", "string[pyarrow]"),
+        ("ActionGeo_Type", "Int64"),
+        ("ActionGeo_Fullname", "string[pyarrow]"),
+        ("ActionGeo_CountryCode", "string[pyarrow]"),
+        ("ActionGeo_ADM1Code", "string[pyarrow]"),
+        ("ActionGeo_Lat", "Float64"),
+        ("ActionGeo_Long", "Float64"),
+        ("ActionGeo_FeatureID", "string[pyarrow]"),
+        ("DATEADDED", "Int64"),
+        ("SOURCEURL", "string[pyarrow]"),
+    ]
+)
+
+
+def test_from_csv_to_parquet(from_csv_to_parquet_client, s3_factory, s3_url):
+    s3 = s3_factory(anon=True)
+    files = s3.ls("s3://gdelt-open-data/events/")[:1000]
+    files = [f"s3://{f}" for f in files]
+
+    df = dd.read_csv(
+        files,
+        sep="\t",
+        names=SCHEMA.keys(),
+        # 'dtype' and 'converters' cannot overlap
+        dtype={
+            col: dtype for col, dtype in SCHEMA.items() if dtype == "string[pyarrow]"
+        },
+        storage_options=s3.storage_options,
+        on_bad_lines="skip",
+        # Some bad files have '#' in numeric values
+        converters={
+            col: lambda v: float(v.replace("#", "") or "NaN")
+            for col, dtype in SCHEMA.items()
+            if dtype != "string[pyarrow]"
+        },
+    )
+
+    # Now we can safely convert the numeric columns
+    df = df.astype(
```

Comment:
it looks like we have pandas 2.0 in the benchmarks now, is there a reason for not using the config to use pyarrow strings?
```python
dask.config.set({"dataframe.convert-string": True});
```

Reply:
We could for sure, but given the explicit `SCHEMA` and the required `converters` to parse strings with erroneous `#` in float fields in certain files followed by then explicitly casting them to their correct dtypes afterwards; does setting this config offer any additional benefit? (I genuinely don't know in this case)

Reply:
Oh, or maybe that string being passed to the converter function is then a pyarrow string..

Reply:
TBH I'm not sure, probably @jrbourbeau or @j-bennet will know better here.

Reply:
Just in case it does: https://github.com/coiled/benchmarks/pull/841/commits/db73e85e9d9adef68ae179ec03899868e0872f0f

Reply:
If you already read your data with correct dtypes, you would not need to do `astype`. Lines 131-133 become unnecessary.

How do you read your data with correct dtypes? The converters should return a correct dtype:

```
converters = {}
for field, dtype in SCHEMA.items():
    if dtype == "Int64":
        converters[field] = lambda x: int(x.replace("#", "") or "NaN")
    elif dtype == "Float64":
        converters[field] = lambda x: float(x.replace("#", "") or "NaN")
df = dd.read_csv("data.csv", converters=converters)
```

Reply:
This would produce 'int64' and 'float64'  not 'Int64' and 'Float64' dtypes, and feel like this is maybe splitting hairs a bit at this point. `astype` is a very common operation and seems like, if anything, would be beneficial to have in the workflow anyhow.

Reply:
True.